### PR TITLE
Adding EntityExists function to RenderSystem

### PR DESCRIPTION
**Pull request by umarquez**
This function will be used to check if an entity exists (and where) into the entities slice, this will prevents entity's duplicity in the system that can causes a double render process for some entities if these was added twice.

This function could also be called from the Remove function to get the position of the entity that will be remove from the system.

### CODE REVIEW

File: common/render.go [hunk 1]
```diff
@@ -136,44 +136,58 @@ func addCameraSystemOnce(w *ecs.World) {
...
-			render.shader = LegacyShader
-		case Rectangle:
-			render.shader = LegacyShader
-		case ComplexTriangles:
-			render.shader = LegacyShader
-		case Text:
-			render.shader = TextShader
-		default:
-			render.shader = DefaultShader
+	if rs.EntityExists(basic) == -1 {
```
Comment (EtienneBruines) [inline common/render.go]: [Idiomatic Go](https://talks.golang.org/2013/bestpractices.slide#2) would be to put the exceptional case on the top (with a return), and then the 'regular logic' behind it. 

```go
if rs.EntityExists(basic) >= 0 {
    return
}

// Setting default shader
// ...
```

File: common/render.go [hunk 2]
```diff
@@ -136,44 +136,58 @@ func addCameraSystemOnce(w *ecs.World) {
...
-		case Text:
-			render.shader = TextHUDShader
-		default:
-			render.shader = HUDShader
+// EntityExists looks if the entity is already into the System's entities. It will return the index >= 0 of the object into de rs.entities or -1 if it could not be found.
+func (rs *RenderSystem) EntityExists(basic *ecs.BasicEntity) int {
+	for index, entity := range rs.entities {
+		if entity.ID() == basic.ID() {
+			return index
+			break
```
Comment (EtienneBruines) [inline common/render.go]: `break` is not needed after `return`, because we're already stopping executing the loop (due to `return`).

**Comment by umarquez** [comment]
@EtienneBruines Fixed!

**Comment by Noofbiz** [comment]
Thank you!

**Pull request merged**